Sources/Commands/Describe.swift

```diff
@@ -166,7 +167,10 @@ fileprivate struct DescribedPackage: Encodable {
             self.path = target.sources.root.relative(to: package.path).pathString
             self.sources = target.sources.relativePaths.map{ $0.pathString }
             self.resources = target.resources.isEmpty ? nil : target.resources
-            self.targetDependencies = target.dependencies.isEmpty ? nil : target.dependencies.compactMap{ $0.target?.name }
+            let targetDependencies = target.dependencies.compactMap{ $0.target }
+            self.targetDependencies = targetDependencies.isEmpty ? nil : targetDependencies.map{ $0.name }
+            let productDependencies = target.dependencies.compactMap{ $0.product }
+            self.productDependencies = productDependencies.isEmpty ? nil : productDependencies.map{ $0.name }
```

Comment:
cc @yim-lee for potential impact on collection generator

Reply:
In theory this should have no impact since it's additive, but that's a good point.  I guess this would have the effect of omitting the `target_dependencies` empty array for a target that had only product dependencies, in addition to adding the `product_dependencies` array.  I don't know if product dependencies is something that you might want to start including in the collections at some point — I don't think that's in the data structures now (are target dependencies there now?).

Reply:
👍 thanks.

The generator is not using `describe` yet (but will), so this change has no impact. I don't think we will include `*_dependencies` in the package collection data, but IIRC the intention is that it will help determine which targets to include (i.e., only targets included in a product).

Reply:
Sounds good.  That's in the separate `product_memberships` array, and unaffected by this.